### PR TITLE
Force instance name to be defined in vagrant deploy

### PR DESCRIPTION
**Pull request by c-bordon**
Close https://github.com/wazuh/wazuh-qa/issues/5311

The PR aims to validate the configuration of an instance name in vagrant, either with the `--instance-name` or `--label-issue` parameter

Tests:

**Vagrant local:** https://github.com/wazuh/wazuh-qa/issues/5311#issuecomment-2096534037
**Issue format validation:** https://github.com/wazuh/wazuh-qa/issues/5311#issuecomment-2096536806
**AWS:** https://github.com/wazuh/wazuh-qa/issues/5311#issuecomment-2096546798
**Vagrant macOS ARM:** https://github.com/wazuh/wazuh-qa/issues/5311#issuecomment-2096577791
**Instance dir concurrency:** https://github.com/wazuh/wazuh-qa/issues/5311#issuecomment-2096606306

### CODE REVIEW

**Review by teddytpc1 — APPROVED** [review]

**Review by teddytpc1 — APPROVED** [review]

File: deployability/modules/allocation/vagrant/provider.py [hunk 1]
```diff
@@ -47,12 +48,28 @@ def _create_instance(cls, base_dir: Path, params: CreationPayload, config: Vagra
         """
         cls.validate_dependencies(params.composite_name)
         if params.instance_name:
-            instance_id = params.instance_name
+            name = params.instance_name
+        elif params.label_issue:
+            issue = params.label_issue
+            url_regex = "(https:\/\/|http:\/\/)?[github]{2,}(\.[com]{2,})?\/wazuh\/[a-zA-Z0-9_-]+(?:-[a-zA-Z0-9_-]+)?\/issues\/[0-9]{2,}"
+            if not re.match(url_regex, issue):
+                raise ValueError(f"The issue label was not provided or is of incorrect format, example: https://github.com/wazuh/<repository>/issues/<issue-number>")
```
Comment (pro-akim) [inline deployability/modules/allocation/vagrant/provider.py:56]: ## Review Notes
Please add log error message in order to debug easily

File: deployability/modules/allocation/vagrant/provider.py [hunk 2]
```diff
@@ -61,10 +78,14 @@ def _create_instance(cls, base_dir: Path, params: CreationPayload, config: Vagra
...
         if platform == 'macos':
             host_instance_dir = "/Users/jenkins/testing/" + instance_id
-            logger.debug(f"Creating instance directory on remote host")
-            cmd = f"mkdir {host_instance_dir}"
             remote_host_parameters = cls.__remote_host(arch, 'create')
             host_identifier = remote_host_parameters['host_provider']
+            logger.debug(f"Checking if instance directory exists on remote host")
+            cmd = f"ls {host_instance_dir} > /dev/null 2>&1"
+            if VagrantUtils.remote_command(cmd, remote_host_parameters):
+                raise ValueError(f"Instance directory {host_instance_dir} already exists on remote host. Check if it is possible to delete the directory on the remote host.")
```
Comment (pro-akim) [inline deployability/modules/allocation/vagrant/provider.py:86]: ## Review Notes
Please add log error message in order to debug easily

**Review by pro-akim — CHANGES_REQUESTED** [review]
## Review Notes
To facilitate debugging in case of failure, add a log message at:

**Review by pro-akim — APPROVED** [review]
## Review Notes
LGTM

**Pull request merged**